### PR TITLE
RM - port derived table invoice_lines_fund_distributions for metadb

### PR DESCRIPTION
**Pull request by sbeltaine**

### CODE REVIEW

**Review by jenmawe — CHANGES_REQUESTED** [review]
- [x] Query runs
- [x] Data output makes sense
- [x] Data types look correct
- [x] Query logic seems correct
- [x] Column names make sense and are spelled correctly
- [x] Query syntax and style are appropriate (e.g., correct use of double quotes, lower case for data types)

For derived tables:
- [x] query has indexes on columns
- [x] vacuum analyze has been added to the bottom
- [] table appears in runlist.txt at an appropriate point

**Review by lib-eric — APPROVED** [review]
@sbeltaine No change since I last approved.

Did you mean to re-request review from @jenmawe ?

I'm guessing just need to double check the other PR for the runlist update. Little confusing since two different requests.

File: sql_metadb/derived_tables/invoice_lines_fund_distributions.sql [hunk 1]
```diff
@@ -0,0 +1,49 @@
+--invoice_lines_fund_distributions
```
Comment (nassibnassar) [inline sql_metadb/derived_tables/invoice_lines_fund_distributions.sql:1]: Missing description: https://github.com/folio-org/folio-analytics/issues/596

File: sql_metadb/derived_tables/invoice_lines_fund_distributions.sql [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+CREATE TABLE invoice_lines_fund_distributions AS
+WITH funds_distr AS (
+    SELECT
+        id AS invoice_line_id,
+        jsonb_extract_path_text(dist.data, 'code') AS fund_distribution_code,
+        jsonb_extract_path_text(dist.data, 'fundId')::uuid AS fund_distribution_id,
+        jsonb_extract_path_text(dist.data, 'distributionType') AS fund_distribution_type,
+        jsonb_extract_path_text(dist.data, 'value')::numeric AS fund_distribution_value,
+        jsonb_extract_path_text(lines.jsonb, 'subTotal')::numeric(12,2) AS invoice_line_sub_total,
+        jsonb_extract_path_text(lines.jsonb, 'total')::numeric(12,2) AS invoice_line_total
```
Comment (nassibnassar) [inline sql_metadb/derived_tables/invoice_lines_fund_distributions.sql:14]: Please note that monetary values for derived tables in Metadb should be `numeric(19,4)`.  I will make the change here.

File: sql_metadb/derived_tables/invoice_lines_fund_distributions.sql [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+DROP TABLE IF EXISTS invoice_lines_fund_distributions;
+
+CREATE TABLE invoice_lines_fund_distributions AS
+WITH funds_distr AS (
+    SELECT
+        id AS invoice_line_id,
+        jsonb_extract_path_text(dist.data, 'code') AS fund_distribution_code,
+        jsonb_extract_path_text(dist.data, 'fundId')::uuid AS fund_distribution_id,
+        jsonb_extract_path_text(dist.data, 'distributionType') AS fund_distribution_type,
+        jsonb_extract_path_text(dist.data, 'value')::numeric AS fund_distribution_value,
```
Comment (nassibnassar) [inline sql_metadb/derived_tables/invoice_lines_fund_distributions.sql:12]: Please note that monetary values for derived tables in Metadb should be `numeric(19,4)`.  I will make the change here.

File: sql_metadb/derived_tables/invoice_lines_fund_distributions.sql [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        jsonb_extract_path_text(lines.jsonb, 'subTotal')::numeric(12,2) AS invoice_line_sub_total,
+        jsonb_extract_path_text(lines.jsonb, 'total')::numeric(12,2) AS invoice_line_total
+    FROM
+        folio_invoice.invoice_lines AS lines
+        CROSS JOIN jsonb_array_elements(jsonb_extract_path(lines.jsonb, 'fundDistributions')) AS dist(data)
```
Comment (nassibnassar) [inline sql_metadb/derived_tables/invoice_lines_fund_distributions.sql:17]: I'm not sure this is correct.  I will try to fix it.

File: sql_metadb/derived_tables/invoice_lines_fund_distributions.sql [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    fund_distribution_id AS fund_distribution_id,
+    ff.fund_status,
+    ff.code AS fund_code,
+    ff.name AS fund_name,
+    ft.id AS fund_type_id,
+    ft.name AS fund_type_name,
+    fund_distribution_value,
+    fund_distribution_type,
+    invoice_line_sub_total,
+    invoice_line_total
```
Comment (nassibnassar) [inline sql_metadb/derived_tables/invoice_lines_fund_distributions.sql:30]: It is probably a good idea to qualify column names with the table name or alias when multiple tables are involved.

Also I wonder about the order of these columns.  For clarity we usually place non-array columns from the base tables before array columns, followed by ordinality, though these are not requirements.

**Comment by nassibnassar** [comment]
I think it might be a good idea to spend some more time with this query before merging.

File: sql_metadb/derived_tables/invoice_lines_fund_distributions.sql (hunk 4 above)
Comment (nassibnassar) [reply sql_metadb/derived_tables/invoice_lines_fund_distributions.sql:17]: Correction, this may be OK.

**Comment by nassibnassar** [comment]
@sbeltaine This PR may need more work.

**Review by nassibnassar — CHANGES_REQUESTED** [review]
At least one change is requested in comments

**Pull request merged**